test/integration/multiplexed_integration_test.cc

```diff
@@ -1944,6 +1944,35 @@ class Http2FrameIntegrationTest : public testing::TestWithParam<FrameIntegration
   }
 };
 
+TEST_P(Http2FrameIntegrationTest, UpstreamRemoteEndstreamWith1xxHeader) {
+  config_helper_.addConfigModifier(
+      [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) -> void { hcm.set_proxy_100_continue(true); });
+  beginSession();
+  FakeRawConnectionPtr fake_upstream_connection;
+
+  // Start a request and wait for it to reach the upstream.
+  sendFrame(Http2Frame::makeRequest(1, "host", "/path/to/long/url"));
+  ASSERT_TRUE(fake_upstreams_[0]->waitForRawConnection(fake_upstream_connection));
+  const Http2Frame settings_frame = Http2Frame::makeEmptySettingsFrame();
+  ASSERT_TRUE(fake_upstream_connection->write(std::string(settings_frame)));
+
+  test_server_->waitForGaugeEq("cluster.cluster_0.upstream_rq_active", 1);
+
+  // A malformed frame with 103 header and END_STREAM is sent, and should not crash Envoy.
+  const std::vector<uint8_t> header_frame =
+      Hex::decode("0000050132000000012dfeff0110000005090d000000010903313033");
```

Comment:
nit: explain how you created this binary

Reply:
Thanks for the quick review, done.